### ISSUE TITLE
0.8.3 permissions on iOS

### ISSUE THREAD

**Issue by eszoke** [blake502/balatro-mobile-maker#67]
Using the new BMM build to make an IPA, I'm getting a permissions error when I go to install it in Altstore. Going back to 0.8.2 to make an IPA with the same .exe and installs fine. Anyone else seeing similar on 0.8.3?

**Comment by blake502** [comment]
Hmm the Python dependency was removed in 0.8.3, based on this PR https://github.com/blake502/balatro-mobile-maker/pull/61#issuecomment-2094901151
I never tested it myself since I don't have iOS. If others are experiencing the same issue, then I'll roll back that change.

**Comment by kylefmohr** [comment]
@eszoke can you share the error message? What OS are you building it on?

**Comment by kylefmohr** [comment]
![IMG_F30138425D92-1](https://github.com/blake502/balatro-mobile-maker/assets/6644803/ed61ba36-f029-433e-819e-19e837fa133f)

Just tested and I'm unable to recreate, installation via AltStore worked just fine for me.

**Comment by eszoke** [comment]
Cool, yeah did a little bit of scenario testing with it but will try some more when I get home a little later. OS was 17.4.1 I believe

**Comment by eszoke** [comment]
Ok tested again with 0.8.3 and the default exe from the current 1.0.1f Balatro build, as well as with an older build 1.0.1c, and both are giving me this error:
![IMG_0021.jpg](https://github.com/blake502/balatro-mobile-maker/assets/1644579/47d375ff-8760-4242-a2d2-0623be414cee)

![IMG_0022.jpg](https://github.com/blake502/balatro-mobile-maker/assets/1644579/e84f965b-db13-4187-ad60-c8810c174a28)

And then tested 0.8.2 with the current 1.0.1f exe again and was able to side load without error.

OS is iPad OS 17.4.1 on my iPad 6. The BMM parameters were 60FPS cap, yes to widescreen/high DPI patches and no to the CRT shader patch.

Super weird yeah, I'll keep playing around with it too might just be something weird with Altstore.

**Comment by kylefmohr** [comment]
That's interesting! I'm not familiar with that error message, but I have a few ideas I'm going to look into this evening. I think the issue stems from the limitations Apple puts on free Developer accounts, which, as you might know, limits you to 3 sideloaded apps on your device at the same time. However including AltStore and the old version of Balatro, this should still leave you with a free slot, so it wouldn't completely explain this error. 

Have you tried uninstalling the old Balatro before installing the new one via AltStore? 

If that still doesn't work, it may work in a few days, when the week-long validity of the apps you signed expires. 

I still want to look into how BMM's IPA base file assigns an App ID.

**Comment by blake502** [comment]
Thanks for looking into this @kylefmohr !
I'll assign the issue to you if you're okay with that.

**Comment by wygoda** [comment]
I'm facing the same issue, building on Windows using BMM 0.8.3, targeting iPhone with iOS 17.4.1 via AltStore 1.7. Works fine with BMM 0.8.2. Attaching a screenshot with more details about the error.
![IMG_5359 - Copy](https://github.com/blake502/balatro-mobile-maker/assets/9084153/5db923a6-eb72-4c21-a453-ec2a95750735)

**Comment by kylefmohr** [comment]
> I'm facing the same issue, building on Windows using BMM 0.8.3, targeting iPhone with iOS 17.4.1 via AltStore 1.7. Works fine with BMM 0.8.2. Attaching a screenshot with more details about the error. Hope it helps! ![IMG_5359 - Copy](https://private-user-images.githubusercontent.com/9084153/328558130-5db923a6-eb72-4c21-a453-ec2a95750735.PNG?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxMzcxNDQsIm5iZiI6MTcxNTEzNjg0NCwicGF0aCI6Ii85MDg0MTUzLzMyODU1ODEzMC01ZGI5MjNhNi1lYjcyLTRjMjEtYTQ1My1lYzJhOTU3NTA3MzUuUE5HP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDUwOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA1MDhUMDI1NDA0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTJjNGU0ODA1N2EwMzk3ZTNiODBhMTM2MDIzYzQ3NGNhMjIzOWZhMDhiMTE0MDFiNDA5MmU0YzMxNDM0NWMzOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.0zwaC8fuQsWKr_tEdr5i_RWp28fzgECZHnIx3lvtz4g)

Thanks for posting the error, I think I see what the issue is, I'll post a new build in a little bit once I figure out a good way to implement this.

**Comment by kylefmohr** [comment]
@eszoke did you also initially run BMM on Windows to build the .ipa file? We might be running into an incompatibility that I'm not sure can be fixed:

tl;dr game.love doesn't have the 'execute' file attribute, and there isn't a good way to do this that I know of on Windows.

I started working on a potential fix for the issue, however [.NET's "File.SetUnixFileMode"](https://learn.microsoft.com/en-us/dotnet/api/system.io.file.setunixfilemode?view=net-8.0) doesn't support Windows either, so we might be out of luck. It may be the case that either Linux or OSX is required to create the .ipa file. Not 100% sure on this, but this is what I'm thinking so far.

**Comment by eszoke** [comment]
Yup I did on Windows, that makes sense though. I have a dual boot of Linux so I'll try creating the IPA file there tomorrow, it would be a perfectly solid solution for me! Cheers

**Comment by Gargly** [comment]
@kylefmohr initially run BMM on Windows to build the .ipa file, but got the same Balatro permission issue. Curious if you know if Linux or OSX are still the way to go for now?

**Comment by kylefmohr** [comment]
> @kylefmohr initially run BMM on Windows to build the .ipa file, but got the same Balatro permission issue. Curious if you know if Linux or OSX are still the way to go for now?

@Gargly Yep, either OSX or Linux should work, personally I’ve tested OSX more

**Comment by ZXSwire3** [comment]
> @kylefmohr initially run BMM on Windows to build the .ipa file, but got the same Balatro permission issue. Curious if you know if Linux or OSX are still the way to go for now?

I was having the same issue on Windows, but was able to get it working on OSX. I'm guessing that you can just put the Balatro OSX app in the same directory as the BMM and it should work, but I used the Balatro.exe from my Windows desktop to make sure it worked since some other people had said that you're supposed to put the executable in the same directory as BMM for it to work.

**Comment by v0idp** [comment]
I have the same issue sadly, nothing seems to work for now. no Mac available here.

**Comment by WarmMarbles** [comment]
Had the same issue, so I used the Linux version on Windows by using WSL. Works like a charm !

**Comment by paulohenriqueaf** [comment]
I was getting the same issue, NSCocoaErrorDomain 513 on 0.8.3 using Windows 11

Works fine with 0.8.2
